src/slapquant/assign.py

```diff
@@ -366,6 +369,15 @@ def identify_UTRs(
                                 cds.end = cds.end + best_offset
                                 utr.start = utr.start + best_offset
 
+                if cds_shortened: cdss_shortened += 1
+                if cds_lengthened: cdss_lengthened += 1
+                if cds_shortened and cds_lengthened:
+                    logger.error(
+                        f"{mRNA.attributes['ID']} had its CDS both "
+                        "shortened and lengthened during UTR assignment."
+                    )
+                    raise ValueError()
+
```

Comment:
I wouldn't use both `logger.error` and `raise ValueError` here. `logger.error` is really only meant for errors you can recover from, such as temporarily unavailable resources, etc. I'd simply say
```python
raise ValueError(f"{mRNA.attributes['ID']} had its CDS both shortened and lengthened during UTR assignment.")
```